flocker/ca/_ca.py

```diff
@@ -0,0 +1,225 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Low-level logic for a certificate authority.
+
+Uses ECDSA + SHA256? Or if that turns out to be too difficult RSA 4096-bit
++ SHA 256.
+"""
+
+import os
+
+from OpenSSL import crypto
+from pyrsistent import PRecord, field
+from twisted.internet.ssl import DistinguishedName, KeyPair, Certificate
+
+
+EXPIRY_20_YEARS = 60 * 60 * 24 * 365 * 20
+
+
+class CertificateAlreadyExistsError(Exception):
+    """
+    Error raised when a certificate file already exists.
+    """
+
+
+class KeyAlreadyExistsError(Exception):
+    """
+    Error raised when a keypair file already exists.
+    """
+
+
+class PathError(Exception):
+    """
+    Error raised when the directory for certificate files does not exist.
+    """
+
+
+class FlockerKeyPair(object):
+    """
+    KeyPair with added functionality for comparison and signing a request
+    object with additional extensions for generating a self-signed CA.
+    """
+    def __init__(self, keypair):
```

Comment:
This seems like a good candidate for `pyrsistent.PRecord` or `characteristic.attributes`. 

If there's a specific reason for not using those, eg eventual upstreaming to twisted, then add a comment to that effect with a Twisted ticket link.

Or perhaps we eventually want users to be able to install flocker-ca with minimal dependencies?